autorest/preparer_test.go

```diff
@@ -713,6 +713,20 @@ func TestWithHeaderAllocatesHeaders(t *testing.T) {
 	}
 }
 
+func TestNestedWithHeaderAllocatesHeaders(t *testing.T) {
+
+	r, err := Prepare(mocks.NewRequest(), Nest(WithHeader("x-foo", "bar"), WithHeader("x-foo2", "bar2")))
```

Comment:
There is some irony in the fact that this actually would work without the `Nest` function as well :)

``` Go
r, err := Prepare(mocks.NewRequest(), WithHeader("x-foo", "bar"), WithHeader("x-foo2", "bar2")))
```

Given that the two do exactly the same thing, does it really make sense to add this function?

I'd love to hear your thoughts! I get the sense that there's just something I'm missing. :)